cdc/cdc.cc

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <utility>
+
+#include "cdc/cdc.hh"
+#include "database.hh"
+#include "db/config.hh"
+#include "dht/murmur3_partitioner.hh"
+#include "schema.hh"
+#include "schema_builder.hh"
+#include "service/migration_manager.hh"
+#include "service/storage_proxy.hh"
+#include "service/storage_service.hh"
+
+using seastar::sstring;
+
+namespace cdc {
+
+static sstring log_name(const sstring& table_name) {
+    static constexpr const auto cdc_log_suffix = "_scylla_cdc_log";
```

Comment:
`const` not needed :)

Reply:
Not sure. I believe `constexpr auto cdc_log_suffix` will be equivalent to `constexpr char* const cdc_log_suffix` here while `constexpr const auto cdc_log_suffix` is `constexpr const char* const cdc_log_suffix`. So `constexpr` makes a pointer const but `constexpr const` makes both the pointer and the element const. I may be wrong though.

Reply:
`const auto x = y;` is equivalent to `auto const x = y;`. If `auto` deduces to `char*` say, then it will become `char* const x = y;`. It will make the pointer const.

What you describe would be achieved with `constexpr const auto* x = y;`. Then when `y` is of type `char*`, auto would deduce to `char`.

Reply:
Fair enough. It seems with auto the const is not necessary, indeed. Without auto it would make a difference:
```
    constexpr const auto a = "_scylla_cdc_log";
    constexpr auto b = "_scylla_cdc_log";
    constexpr const char* c = "_scylla_cdc_log";
    constexpr char* d = "_scylla_cdc_log";
    std::cerr << demangle(typeid(a).name()) << std::endl;
    std::cerr << demangle(typeid(b).name()) << std::endl;
    std::cerr << demangle(typeid(c).name()) << std::endl;
    std::cerr << demangle(typeid(d).name()) << std::endl;
```

```
char const*
char const*
char const*
char*
```

Reply:
"With `auto`" it would also make a difference, but only if you used `auto*`. I.e.
`constexpr auto* e = "_scylla_cdc_log";` would make `e` have type `char*`.